Automattic/jetpack
# Related Posts: Add per-block header settings.

Fixes #30898

## Proposed changes:


Adds per-block headlines to the Related Posts block. The updated UI in the block editor looks like this:

<img width="1004" alt="SCR-20230524-smbz" src="https://github.com/Automattic/jetpack/assets/40267301/0d196efe-800e-49ce-8380-14ef914a0178">

### Other information:


## Jetpack product discussion



pejTkB-yT-p2

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






1. Spin up a test site with this branch.
3. Make sure Related Posts are enabled via Jetpack → Settings → Traffic → Related Posts.
4. Create a new post and add a Related Posts block.
5. Confirm the "Display headline" toggle is present and disabled by default.
6. Confirm enabling the toggle shows the text entry field.
7. Confirm entering a headline updates the preview of the block.
8. Confirm the headline is shown on the front-facing page once published (if not left empty).

**Note:** You may need to publish before the placeholders are replaced with actual post content.
